src/ASM/FiniteElement.h

```diff
@@ -14,6 +14,7 @@
 #ifndef _FINITE_ELEMENT_H
 #define _FINITE_ELEMENT_H
 
+#include "BasisFunctionVals.h"
```

Comment:
OK, but then you can remove the forward declaration of the same type in L23 (not shown here)

Reply:
done.